### PR TITLE
Fixed #31282 -- Corrected RelatedManager docs for using add/remove/set with PKs.

### PR DESCRIPTION
**Pull request by carltongibson**
Using add/remove/set with PKs is only valid for M2M.
Added tests showing errors for FK.

Regression in a44a21a22f20c1a710670676fcca798dd6bb5ac0

### CODE REVIEW

File: tests/many_to_one/tests.py [hunk 1]
```diff
@@ -733,3 +733,13 @@ def test_reverse_foreign_key_instance_to_field_caching(self):
         child = parent.to_field_children.get()
         with self.assertNumQueries(0):
             self.assertIs(child.parent, parent)
+
+    def test_add_remove_set_by_pk(self):
+        usa = Country.objects.create(name='United States')
+        chicago = City.objects.create(name='Chicago')
+        with self.assertRaisesMessage(TypeError, "'City' instance expected, got 1"):
+            usa.cities.add(chicago.pk)
+        with self.assertRaisesMessage(AttributeError, "'int' object has no attribute '_meta'"):
```
Comment (felixxm) [inline tests/many_to_one/tests.py]: I think we should raise a more descriptive error, maybe the same as in `add()` and `set()`.

File: tests/many_to_one/tests.py [hunk 2]
```diff
@@ -733,3 +733,13 @@ def test_reverse_foreign_key_instance_to_field_caching(self):
         child = parent.to_field_children.get()
         with self.assertNumQueries(0):
             self.assertIs(child.parent, parent)
+
+    def test_add_remove_set_by_pk(self):
+        usa = Country.objects.create(name='United States')
+        chicago = City.objects.create(name='Chicago')
+        with self.assertRaisesMessage(TypeError, "'City' instance expected, got 1"):
```
Comment (felixxm) [inline tests/many_to_one/tests.py]: `1` -> `chicago.pk`

File: tests/many_to_one/tests.py [hunk 3]
```diff
@@ -733,3 +733,13 @@ def test_reverse_foreign_key_instance_to_field_caching(self):
...
             self.assertIs(child.parent, parent)
+
+    def test_add_remove_set_by_pk(self):
+        usa = Country.objects.create(name='United States')
+        chicago = City.objects.create(name='Chicago')
+        with self.assertRaisesMessage(TypeError, "'City' instance expected, got %s" % chicago.pk):
+            usa.cities.add(chicago.pk)
+        with self.assertRaisesMessage(AttributeError, "'int' object has no attribute '_meta'"):
+            usa.cities.remove(chicago.pk)
+        with self.assertRaisesMessage(TypeError, "'City' instance expected, got %s" % chicago.pk):
```
Comment (felixxm) [inline tests/many_to_one/tests.py]: We can move `msg` before assertions
```python
msg = "'City' instance expected, got %s" % chicago.pk
with self.assertRaisesMessage(TypeError, msg):
     ...
with self.assertRaisesMessage(TypeError, msg):
    ...
```

File: tests/many_to_one/tests.py (hunk 1 above)
Comment (carltongibson) [reply tests/many_to_one/tests.py]: Yes, I was a bit surprised...

File: tests/many_to_one/tests.py (hunk 2 above)
Comment (carltongibson) [reply tests/many_to_one/tests.py]: Doh.

File: tests/many_to_one/tests.py (hunk 3 above)
Comment (carltongibson) [reply tests/many_to_one/tests.py]: OK, let me improve `remove()` and I'll adjust.

File: django/db/models/fields/related_descriptors.py [hunk 4]
```diff
@@ -697,6 +697,10 @@ def remove(self, *objs, bulk=True):
                 val = self.field.get_foreign_related_value(self.instance)
                 old_ids = set()
                 for obj in objs:
+                    if not isinstance(obj, self.model):
+                        raise TypeError("'%s' instance expected, got %r" % (
+                            self.model._meta.object_name, obj,
+                        ))
```
Comment (hramezani) [inline django/db/models/fields/related_descriptors.py]: Shouldn't this new error add in a separate commit?

Comment (carltongibson) [reply django/db/models/fields/related_descriptors.py]: So currently there are two commits: one for the docs change. One for clarifying the behaviour (which adds this error and the tests.) 

🤷‍♂

**Pull request merged**